### DISCUSSION TITLE
NO `blocking` when `fallback: true` via `next/link`?

### DISCUSSION THREAD

**Discussion by AssisrMatheus** [vercel/next.js#42924] [Help] [answered]
![CleanShot 2022-11-14 at 21 48 59](https://user-images.githubusercontent.com/12276174/201814453-30d9561d-163b-472f-844f-1dc54e2b2224.png)

In my case. I want the exact opposite. I want it to not be blocking and have a visual loading for the user

Is there a way to have NO `blocking` when `fallback: true` via `next/link`?

**Comment by icyJoseph** [comment, marked as answer]
Aha, interesting, because `next/link` triggers data-fetching for GetStaticProps with ISR, at worst on-hover, and at best when the link is in the viewport. You can control this with the `prefetch` prop on `next/link`, so for the most part, by the time a user clicks on a link there's data for it already, so it just transitions into the page. For GSSP data-fetching is always on click.

For all of the above, JS (to run the page) is prefetched.

Of course, a heavy page, or one that collects data from various sources might take longer than any prefetching could optimise for, in that case, `Router.events` should help you out, I guess you gotta listen for:

> - routeChangeStart(url, { shallow }) - Fires when a route starts to change
> - routeChangeComplete(url, { shallow }) - Fires when a route changed completely

And check that the url path is going into the page where you want to show a spinner.

The part where it gets tricky is that you need to control a Spinner from within the callback passed to these events. For example, https://caspertheghost.me/blog/nprogress-next-js#router-events, which uses NProgress.

Note that, fallback, only happens when there's no ISR'd (at build time or runtime) page available.
